File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        int carsPerHour = 221;
        double production = 0;

        if (speed >=0 && speed <=4){
            production = (carsPerHour * speed);
        }else if (speed >4 && speed <=8){
            production = (carsPerHour * speed) * 0.90;
        }else if (speed == 9){
            production = (carsPerHour * speed) * 0.80;
        }else if (speed >= 10){
            production = (carsPerHour * speed) * 0.77;
        }
        return production;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed) / 60;
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    static int production = 221;
    
    public static double SuccessRate(int speed)
    {
        if (speed >=1 && speed <=4){
            return 1.00;
        }else if (speed >=5 && speed <=8){
            return 0.90;
        }else if (speed == 9){
            return 0.80;
        }else if (speed == 10){
            return 0.77;
        }else{
            return 0;
        }
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        return ((double)speed * (double)production) * SuccessRate(speed);
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)ProductionRatePerHour(speed)/60;
    }
}
